### PR TITLE
kubetest2-gke: support acquiring project from boskos

### PR DESCRIPTION
**Pull request by cofyc**
I am planning to use `kubetest2` in the prow environment.  Boskos support is required.

It includes:

- move boskos to a separate pkg
- support boskos in kubetest2-gke
- register klog flags in kubetest2-gke

### CODE REVIEW

**Comment by cofyc** [comment]
/assign @amwat
PTAL

File: kubetest2/pkg/boskos/boskos.go [hunk 1]
```diff
@@ -44,36 +45,34 @@ func makeBoskosClient(boskosLocation string) (*client.Client, error) {
 	return boskos, nil
 }
 
-// getProjectFromBoskos creates a boskos client, acquires a gcp project
-// and starts a heartbeat goroutine to keep the project reserved
-func getProjectFromBoskos(boskosClient *client.Client, timeout time.Duration, heartbeatClose chan struct{}) (string, error) {
-	resourceType := "gce-project"
+// Acquire acquires a resource for the given type and starts a heartbeat goroutine to keep the resource reserved.
+func Acquire(boskosClient *client.Client, resourceType string, timeout time.Duration, heartbeatClose chan struct{}) (*common.Resource, error) {
```
Comment (cofyc) [inline kubetest2/pkg/boskos/boskos.go:49]: I also renamed these methods to make them generic.

**Comment by amwat** [comment]
We just moved to a new repo (I was about to file a notice in the readme here 😄  )
Do you want to send a PR there directly?
https://github.com/kubernetes-sigs/kubetest2

**Comment by cofyc** [comment]
> We just moved to a new repo (I was about to file a notice in the readme here 😄 )
> Do you want to send a PR there directly?
> https://github.com/kubernetes-sigs/kubetest2

Great!

**Comment by cofyc** [comment]
Opened a PR in kubetest2 repo: https://github.com/kubernetes-sigs/kubetest2/pull/3

**Pull request closed without merging**